File: Sources/ComprovacaoFacilLattes/Models/LattesSection.swift
import Foundation
import SwiftData

@Model
final class LattesSection {
    @Attribute(.unique) var id: UUID
    var title: String
    var order: Int

    var profile: LattesProfile?

    @Relationship(deleteRule: .cascade, inverse: \LattesEntry.section)
    var entries: [LattesEntry]

    init(title: String, order: Int) {
        self.id = UUID()
        self.title = title
        self.order = order
        self.entries = []
    }

    var sortedEntries: [LattesEntry] {
        entries.sorted { $0.order < $1.order }
    }

    var confirmedCount: Int { entries.filter { $0.certificateStatus == .confirmed }.count }
    var suggestedCount: Int { entries.filter { $0.certificateStatus == .suggested }.count }
    var pendingCount: Int   { entries.filter { $0.certificateStatus == .none }.count }
}

File: Versão Windows/src/ComprovacaoFacilLattes.Core/Models/LattesSection.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ComprovacaoFacilLattes.Core.Models;

/// <summary>
/// Uma seção do currículo — "Artigos completos…", "Participação em bancas… - Mestrado", etc.
/// O parser SUBDIVIDE seções do Lattes em várias seções do app; <see cref="Title"/> já inclui
/// sufixos de subdivisão, ex. "Atuação profissional - UFAC - Disciplinas ministradas".
/// </summary>
public class LattesSection
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Title { get; set; } = "";
    public int Order { get; set; }

    public Guid? ProfileId { get; set; }
    public LattesProfile? Profile { get; set; }

    public List<LattesEntry> Entries { get; set; } = new();

    public LattesSection() { }

    public LattesSection(string title, int order)
    {
        Title = title;
        Order = order;
    }

    [NotMapped]
    public IEnumerable<LattesEntry> SortedEntries => Entries.OrderBy(e => e.Order);

    [NotMapped]
    public int ConfirmedCount => Entries.Count(e => e.CertificateStatus == EntryStatus.Confirmed);

    [NotMapped]
    public int SuggestedCount => Entries.Count(e => e.CertificateStatus == EntryStatus.Suggested);

    [NotMapped]
    public int PendingCount => Entries.Count(e => e.CertificateStatus == EntryStatus.None);
}
